src/main/scala/org/hpi/esb/datasender/Main.scala

```
@@ -1,49 +1,179 @@
+
+  def main(args: Array[String]) = {
+
+    val cliConfig = handleCliArguments(args)
+    val configFiles = getConfigFiles(cliConfig)
+    val fileConfig = handleConfiguration(configFiles)
+
+    val configToUse = mergeCliAndFileConfig(cliConfig, fileConfig)
+
+    if (cliConfig.verbose)
+    {
+      Logging.setToDebug
+      logger.info("DEBUG/VERBOSE mode switched on")
+    }
+    else Logging.setToInfo
+
+    if (! configToUse.verify)
+    {
+      logger.error("Invalid configuration:\n" + configToUse.toString)
+      System.exit(-1)
     }
+
+    val dataProducer = new DataProducer(configToUse.datasender)
+    dataProducer.execute
   }
 
-  val verboseOptionShort:String = "-v"
-  val verboseOptionLong:String = "--verbose"
-  val usage = s"""
-    Usage: sbt run | sbt \"run $verboseOptionShort\"  sbt \"run $verboseOptionLong\"
-  """
+  def isValidFile(path: String) : Boolean = Path.fromString(path).exists && Path.fromString(path).isFile
 
-  def main(args: Array[String]) = {
+  def deleteTmpDefaultConf()
+  {
+    new java.io.File(TempDefaultConf).delete
+  }
+
+  def createTmpDefaultConf() : java.io.File =
+  {
+    val stream = getClass.getResourceAsStream(RelativeDefaultConfigPath)
+    val lines = scala.io.Source.fromInputStream( stream ).getLines.mkString("\n")
+    deleteTmpDefaultConf
+    new PrintWriter(TempDefaultConf) { write(lines); close }
+    new java.io.File(TempDefaultConf)
+  }
+
+  def getConfigFiles(cliConfig: CliConfig) : Seq[java.io.File] =
+  {
+    var configPath: String = ""
+    if (cliConfig.alternativeConfig.isDefined)
+      configPath = cliConfig.alternativeConfig.get
+    else
+      configPath = userConfigPath
+
+    var configFiles = Seq[java.io.File]()
+
+    if (isValidFile(configPath))
+    {
+      logger.info("Using config file " + configPath)
+      configFiles = new java.io.File(configPath) +: configFiles
+    }
+    else
+      logger.warn(configPath + " is not a valid file")
 
```

Should we might abort execution at this point in order to fail fast? 
One could probably easily read over such a small warning, which could be the root cause for unexpected behaviour.